themes/10up-theme/includes/blocks.php

```diff
@@ -128,6 +130,52 @@ function blocks_editor_styles() {
 	}
 }
 
+
+/**
+ * Enqueue block specific styles.
+ *
+ * This function is used to enqueue styles that are specific to a block. It
+ * first gets all the CSS files in the 'blocks/autoenqueue' directory. Then
+ * for each stylesheet, it determines the block type by removing the directory
+ * path and '.css' from the stylesheet path. It then tries to get the asset
+ * file for the block type. If the asset file doesn't exist, it creates a new
+ * one with the version set to the file modification time of the stylesheet
+ * and no dependencies. Finally, it enqueues the block style using the block
+ * type, the URL to the stylesheet, the path to the stylesheet, the version
+ * from the asset file, and the dependencies from the asset file.
+ *
+ * @return void
+ */
+function enqueue_block_specific_styles() {
+	$stylesheets = glob( TENUP_THEME_DIST_PATH . '/blocks/autoenqueue/*.css' );
+
+	foreach ( $stylesheets as $stylesheet_path ) {
+		$block_type = str_replace( TENUP_THEME_DIST_PATH . '/blocks/autoenqueue/', '', $stylesheet_path );
+		$block_type = str_replace( '.css', '', $block_type );
+		$asset_file = TENUP_THEME_DIST_PATH . 'blocks/autoenqueue/' . $block_type . '.asset.php';
+
+		if ( ! file_exists( $asset_file ) ) {
+			$asset_file = require $asset_file;
+		} else {
+			$asset_file = [
+				'version'      => filemtime( $stylesheet_path ),
+				'dependencies' => [],
+			];
+		}
+
+		wp_enqueue_block_style(
+			$block_type,
+			[
+				'handle'       => "{$block_type}",
```

Comment:
Another change I'd recommend is changing the handle here to something that is namespace with the theme.

If we could make it so the handle is something like: `namespace-theme-${block_namespace}-${block_name}-style`

Reply:
Resolved in [290815e](https://github.com/10up/wp-scaffold/pull/229/commits/290815e423c7a71afe3f2047a97d3f4bf7045f49)